Sources/Build/BuildDelegate.swift

```diff
@@ -237,6 +236,36 @@ public final class BuildExecutionContext {
     }
 }
 
+public protocol PackageStructureDelegate {
+    func packageStructureChanged() -> Bool
+}
+
+final class PackageStructureCommand: CustomLLBuildCommand {
+
+    override func getSignature(_ command: SPMLLBuild.Command) -> [UInt8] {
+        // Include build parameters and process env in the signature.
+        var hash = Data()
+        do {
+            hash += try self.ctx.buildParameters.hash()
+
+            let encoder = JSONEncoder()
+            if #available(macOS 10.13, *) {
```

Comment:
This code is going to be broken pre 10.13, should we just update our required deployment target? Also, isn't this going to be broken on Linux?

Reply:
SwiftPM anyway won't deploy pre 10.13. I have a PR (#2198) up for bumping the deployment target but need to validate it. Also, this check works on Linux because of `*`.